### DISCUSSION TITLE
Web 前端 AI 抠图应用实践

### DISCUSSION THREAD

**Discussion by giscus** [ceynri/ceynri.cn#64] [Comments]
# Web 前端 AI 抠图应用实践

山风的小角落 / Ceynri's personal website

https://ceynri.cn/blog/frontend-ai-matting/

**Comment by justmorefree** [comment]
作者你好，感谢你的文章，受益匪浅。同时也想请教下，要实现类似Segment Anyting的图片分割效果，鼠标hover时，自动抓去一部分，这个纯前端能实现吗，还是需要后端部署Sam模型？

**Reply by ceynri** [reply]
SAM 模型官方提供的 demo 就是纯前端的图像分割的。而文章中提到的几个图像抠图的模型则不支持由用户输入点击位置去抠指定的图像元素，可以研究一下有没有办法结合😄

**Reply by justmorefree** [reply]
嗯…segment anything demo没看太明白，是需要执行python脚本，获取到模型文件，放到前端项目里，继续后续操作就可以了吗

**Reply by ceynri** [reply]
我没有深入研究 SAM，你可以自行查阅相关的资料，比如 [issue#55](https://github.com/facebookresearch/segment-anything/issues/55)，说明了官方并没有开源完整的前端代码，而只是提供了一个具备基础能力的 mini demo。
不过可以提醒的是，如果你想用前端实现相关能力，一般不会涉及到 Python 脚本相关的处理。AI 模型的使用方式往往会只会说明 Python 的用法，你可能需要寻找其对应的 JS 实现，或者尝试研究如何处理 onnx 格式的模型读取。
在这方面，使用 transformers.js 是最简单入门的方式，他们提供的模型都已经封装好了可以直接用。

**Reply by justmorefree** [reply]
好的，非常感谢

**Comment by q737556848** [comment]
谢谢。学习了。正想往前端ai方向走走。宝藏文章
